spec/models/claim/matching_attribute_finder_spec.rb

```diff
@@ -2,24 +2,41 @@
 
 RSpec.describe Claim::MatchingAttributeFinder do
   describe "#matching_claims" do
-    let(:source_claim) {
-      create(
-        :claim,
+    let(:claim_attributes) {
+      {
         teacher_reference_number: "0902344",
         national_insurance_number: "QQ891011C",
         email_address: "genghis.khan@mongol-empire.com",
         bank_account_number: "34682151",
         bank_sort_code: "972654",
-        building_society_roll_number: "123456789/ABCD"
-      )
+        building_society_roll_number: "123456789/ABCD",
+        policy: StudentLoans
+      }
     }
 
-    let!(:claim_with_no_matching_attributes) { create(:claim, :submitted) }
-    let!(:unsubmitted_claim_with_matching_teacher_reference_number) { create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number) }
+    let(:source_claim) { create(:claim, claim_attributes) }
 
     subject(:matching_claims) { Claim::MatchingAttributeFinder.new(source_claim).matching_claims }
 
-    it "does not include the source claim, or claims that do not match, or claims that are not submitted" do
+    it "does not include the source claim" do
+      expect(matching_claims).to be_empty
+    end
+
+    it "does not include claims that do not match" do
+      create(:claim, :submitted)
+
+      expect(matching_claims).to be_empty
+    end
+
+    it "does not include unsubmitted claims with matching attributes" do
+      create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number)
+
+      expect(matching_claims).to be_empty
+    end
+
+    it "does not include claims that match, but have a different policy" do
+      create(:claim, :submitted, claim_attributes.merge(policy: MathsAndPhysics))
```

Comment:
All the other specs grab an attribute from `source_claim` so this appears to be the only place where `claim_attributes` is used aside from when `source_claim` is created. Which is making me wonder if it's worth it to extract out the attributes. Could this spec not do the same as the others and grab the TRN from source_claim? It would mean one less instance variable to understand and trace through the specs.

Reply:
Yeah, that makes sense, I guess I was trying to be super-clear about the fact that they were from the same person, but it'll show up in the matcher regardless of the number of attributes that match.